### PR TITLE
cli: Add `anchor expand` command

### LINKED ISSUE #1111: cli: anchor expand
1. ship cargo expand with anchor and make it available through `anchor expand` 
2. anchor expand also redirects the output into a file in a directory that we include in .gitignore
3. instead of overwriting existing files in that directory, it creates a new one every time with some informative name. maybe the time , anchor-version etc.

to me this isn't that important and just a nice little ux improvement but maybe there are more reasons this would be a good idea (or wouldn't be)

### PR DESCRIPTION
**Pull request by paul-schaaf**
closes #1111

### CODE REVIEW

**Comment by hwsimmons17** [comment]
This is awesome!

**Comment by paul-schaaf** [comment]
- [ ] adds command description to `commands.md`

**Comment by utx0** [comment]
Oh I was just looking for this exact feature. Nice work. Tested locally and works like a dream.

File: docs/src/cli/commands.md [hunk 1]
```diff
@@ -30,21 +31,6 @@ SUBCOMMANDS:
                Cargo.toml
 ```
 
-## Init
```
Comment (paul-schaaf) [inline docs/src/cli/commands.md:33]: this list is now in alphabetical order like the overview list above

**Comment by paul-schaaf** [comment]
incorrect commit msg for commit #ce2e1fa but code is correct. can be ignored

**Pull request merged**